torch/distributed/distributed_c10d.py

```diff
@@ -1042,13 +1045,13 @@ def get_backend(group: Optional[ProcessGroup] = None) -> str:
         raise ValueError("Invalid process group specified")
     pg_store = _world.pg_map[pg] if pg in _world.pg_map else None
     assert pg_store is not None
-    return pg_store[0]
+    return Backend(pg_store[0])
 
 
 _exception_logger
 @_time_logger
 def init_process_group(
-    backend: Union[str, Backend] = None,
+    backend: Optional[Backend] = None,
```

Comment:
should this be Optional[Union[str, Backend]]?

Reply:
so part of this change was to change Backend to be derived from str in addition to the metaclass magic.  this simplified some of these unions.  I actually though am surprised the type checker accepts this as I think it should be `Optional[str]`

I think @time_logger may be interfering with mypy...

Reply:
yes!  the decorators were interfering.  fixed.  (also notice `_exception_logger` isn't a decorator above... fixing that too, apparently it's always been broken)